### PR TITLE
fix: user can only list models with correct permissions (OSS has appropriate changes) + small fixes in workspace filtering in get models 

### PR DESCRIPTION
**Pull request by nrajanee**
## Description

fix: user can only list models with correct permissions (OSS has appropriate changes) + small fixes in workspace filtering in get models 

## Test Plan

e2e_tests included.

### CODE REVIEW

File: master/internal/api_model.go [hunk 1]
```diff
@@ -142,27 +136,43 @@ func (a *apiServer) GetModels(
...
-			int32(workspaceIDFilterExpr)); err != nil {
-			return nil, err
-		} else if !ok {
-			return nil, errors.Errorf(
-				"current user %q doesn't have view permissions in the given workspace with id: %v.",
-				curUser.Username, workspaceIDFilterExpr)
+	// function below returns a list of workspaces that have permissions
+	// filtered according to user given workspaces.
+	// if global permissions and no filter list given by user then it's an empty list.
+	workspaceIdsWithPermsAndFilterList, ok, err := modelauth.AuthZProvider.Get().
```
Comment (ioga) [inline master/internal/api_model.go:157]: sounds like `workspaceIdsWithPermsAndFilterList` is a list of workspace ids where user have a permission to read models, maybe call it ``workspaceIdsWithPermsList`?

Comment (nrajanee) [reply master/internal/api_model.go:157]: The list is also filtered according to the user given input (if they want to list models in specific workspaces).

**Review by ioga — APPROVED** [review]

**Review by julian-determined-ai — CHANGES_REQUESTED** [review]
I am seeing behavior where the initial filtering works on our Model table on the FE but the subsequent request does not. I cannot recreate this on any other branch but haven't tracked down the cause in this branch yet. Currently still investigating.


https://user-images.githubusercontent.com/103522725/219759563-257534a8-0728-4dc9-8008-0c76bdb9636e.mov

**Comment by nrajanee** [comment]
> I am seeing behavior where the initial filtering works on our Model table on the FE but the subsequent request does not. I cannot recreate this on any other branch but haven't tracked down the cause in this branch yet. Currently still investigating.

@julian-determined-ai hey julian, thank you for checking this. but this check won't work here OSS but makes more sense with the EE code: https://github.com/determined-ai/determined-ee/pull/681. Thank you!

**Comment by julian-determined-ai** [comment]
> > I am seeing behavior where the initial filtering works on our Model table on the FE but the subsequent request does not. I cannot recreate this on any other branch but haven't tracked down the cause in this branch yet. Currently still investigating.
> 
> @julian-determined-ai hey julian, thank you for checking this. but this check won't work here OSS but makes more sense with the EE code: [determined-ai/determined-ee#681](https://github.com/determined-ai/determined-ee/pull/681). Thank you!

Just checking to make sure i understand are you saying once this and the other PR are merged in OSS will behave fine?

**Comment by nrajanee** [comment]
> > > I am seeing behavior where the initial filtering works on our Model table on the FE but the subsequent request does not. I cannot recreate this on any other branch but haven't tracked down the cause in this branch yet. Currently still investigating.
> > 
> > 
> > @julian-determined-ai hey julian, thank you for checking this. but this check won't work here OSS but makes more sense with the EE code: [determined-ai/determined-ee#681](https://github.com/determined-ai/determined-ee/pull/681). Thank you!
> 
> Just checking to make sure i understand are you saying once this and the other PR are merged in OSS will behave fine @nrajanee ?

Talking about this on slack. I misunderstood before. This is something that's not related to EE. Listing models in specific workspaces should work in OSS too. It works on the CLI. @julian-determined-ai and I are investigating. thank you!

**Review by julian-determined-ai — APPROVED** [review]
I am actually unable to reproduce the bug that I posted locally on this branch. This may just be an issue very specific to the Netlify deployment on this branch. 

Changes look good to me.

**Pull request merged**